server/src/main/java/org/elasticsearch/search/fetch/PreloadedFieldLookupProvider.java

```diff
@@ -32,12 +33,18 @@ class PreloadedFieldLookupProvider implements LeafFieldLookupProvider {
 
     @Override
     public void populateFieldLookup(FieldLookup fieldLookup, int doc) throws IOException {
-        String field = fieldLookup.fieldType().name();
+        if (storedFields == null) {
+            storedFields = Collections.emptyMap();
+        }
+        final String field = fieldLookup.fieldType().name();
         if (storedFields.containsKey(field)) {
             fieldLookup.setValues(storedFields.get(field));
             return;
         }
         // stored field not preloaded, go and get it directly
+        if (backUpLoader == null && loaderSupplier == null) {
```

Comment:
I do not expect this to normally happen because we call `setNextReader` while iterating search hit results.

Reply:
I'm ok also not checking for this and then `backUpLoader = loaderSupplier.get();` fails with NPE?

Reply:
Well the thing is it might happen if `setNextReader` is not called for any reason...which should not really be the case.